frontend/src/layouts/EventViewer/EventViewer.tsx

```diff
@@ -10,14 +10,53 @@ import { TraceEvent } from "@/validators/TraceEvent";
 import { Loader2, X } from "lucide-react";
 import { ReactNode, useMemo } from "react";
 import ReactJson, { ThemeObject } from "react-json-view";
+import { Marker } from "react-mark.js";
+import { useRef, useEffect, useState } from "react";
+import { EventSearchCriteria } from "@/hooks/useEventsSearch";
 
 interface EventViewerProps {
   events: TraceEvent[];
   selectedEventCallId: string | null;
   onEventClose: () => void;
   viewerPlaceholder: ReactNode;
+  searchValue: string;
+  searchBy: EventSearchCriteria;
 }
-export const EventViewer = ({ events, selectedEventCallId, onEventClose, viewerPlaceholder }: EventViewerProps) => {
+
+export const EventViewer = ({
+  events,
+  selectedEventCallId,
+  onEventClose,
+  viewerPlaceholder,
+  searchValue,
+  searchBy,
+}: EventViewerProps) => {
+  const [searchValueMarker, setSearchValueMarker] = useState({ searchValue });
+
+  useEffect(() => {
+    setSearchValueMarker({ searchValue: searchValue });
+  }, [searchValue, searchBy]);
+
+  const jsonViewerRef = useRef(null);
+  useEffect(() => {
+    const observer = new MutationObserver((mutationsList) => {
+      if ((mutationsList[0].target as HTMLElement).className === "icon-container") {
+        setSearchValueMarker({ searchValue: searchValue });
```

Comment:
I would have a separate useState for triggering manual rerenders. E.g. from a hook like this one:
```
function useForceUpdate() {
  const [, setToggle] = React.useState(false)
  return () => setToggle(toggle => !toggle)
}
```

Reply:
And change the `useState({ searchValue })` to just `useState(searchValue)`